addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java

```diff
@@ -175,6 +190,77 @@ public OastPanel getOastPanel() {
         return oastPanel;
     }
 
+    /**
+     * @return the selected external OAST service (i.e. excluding Callbacks) for usage in active
+     *     scan rules, or {@code null} if no service is selected.
+     */
+    public OastService getActiveScanOastService() {
+        if (OastParam.NO_ACTIVE_SCAN_SERVICE_SELECTED_OPTION.equals(
+                oastParam.getActiveScanServiceName())) {
+            return null;
+        }
+        return getOastServices().get(oastParam.getActiveScanServiceName());
+    }
+
+    public String registerAlertAndGetPayload(Alert alert) throws Exception {
+        if (getActiveScanOastService() != null) {
+            String payload = getActiveScanOastService().getNewPayload();
+            alertCache.put(payload, alert);
+            return payload;
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void activeScanAlertOastRequestHandler(OastRequest request) {
+        try {
+            HttpMessage oastReceivedMsg = request.getHistoryReference().getHttpMessage();
+            String uri = oastReceivedMsg.getRequestHeader().getURI().toString();
+            if (alertCache.keySet().stream().noneMatch(k -> uri.contains(k.toString()))) {
+                return;
+            }
+            Alert alert = null;
+            for (Object k : alertCache.keySet()) {
+                String key = k.toString();
+                if (uri.contains(key)) {
+                    alert = (Alert) alertCache.get(key);
+                }
+            }
+            if (alert == null) {
+                LOGGER.warn(
+                        "Soft reference to alert object for interaction at {} expired. Not raising alert.",
+                        uri);
+                return;
+            }
+            alert.setOtherInfo(
+                    alert.getOtherInfo()
+                            + "Request\n"
```

Comment:
- Add newline before request too?
- Ideally this should be i18n (same for `Response`).

Reply:
adding a newline before Request is resulting in a lot of additional newlines, so skipping that...